anodizer/anodizer-target-codegen/src/main/kotlin/io/github/amzn/anodizer/codegen/context/Ctx.kt

```diff
@@ -0,0 +1,191 @@
+package io.github.amzn.anodizer.codegen.context
+
+import io.github.amzn.anodizer.AnodizerModel
+import io.github.amzn.anodizer.core.Definition
+import io.github.amzn.anodizer.core.Name
+import io.github.amzn.anodizer.core.Type
+import net.pearx.kasechange.CaseFormat
+import net.pearx.kasechange.toCamelCase
+import net.pearx.kasechange.toPascalCase
+import net.pearx.kasechange.toSnakeCase
+
+/**
+ * The [Ctx] abstraction is used to make a definition tree amenable for code generation.
+ *
+ * Q: Why?
+ * A: A ctx wraps the core anodizer model with code generation specific information/context.
+ *
+ * Q: Why not have the model hold the necessary context?
+ * A: The model is independent of codegen, hence the separation, but that might be a welcome addition to the model.
+ *
+ * Q: The word "ctx" and "context" is generic and overused.
+ * A: Agreed, it's a tree-node decorator pattern but the "context" keyword is familiar to the codegen space.
+ */
+public sealed interface Ctx {
```

Comment:
So, `Ctx` is the moral equivalent of the bit 'data-hash' that is needed for stateless templates like mustache.

Reply:
Yes, some mustache places call it a "context" and others call it the "hash".